src/Tasks/Microsoft.NET.Build.Tasks/AppHost.cs

```diff
@@ -56,9 +53,12 @@ public static class AppHost
             File.Copy(appHostSourceFilePath, appHostDestinationFilePath, overwriteExisting);
 
             // Re-write ModifiedAppHostPath with the proper contents.
-            using (FileStream fs = new FileStream(appHostDestinationFilePath, FileMode.Truncate, FileAccess.ReadWrite, FileShare.Read))
+            using (var memoryMappedFile = MemoryMappedFile.CreateFromFile(appHostDestinationFilePath, FileMode.Open))
```

Comment:
I am not sure about this line of `FileMode.Open`. However, according to the doc. It has to be Open. https://msdn.microsoft.com/en-us/library/dd987972(v=vs.110).aspx

Reply:
The interesting argument for memory mapped access level is actually MemoryMappedFileAccess, which doesn't appear in this overload, but the default (ReadWrite) is what we want.

I would just use CreateFromFile(string) and let it use all defaults. Or, if you want to be explicit about the access, then you could go with:

``` C#
MemoryMappedFile.CreateFromFile(
    appHostDestinationFilePath,
    FileMode.Open, 
    mapName: default,
    capacity: default,
    MemoryMappedFileAccess.ReadWrite);
```

Reply:
Fixed, used CreateFromFile(string)